controller/kubernetes_node_controller.go

```diff
@@ -198,11 +198,17 @@ func (knc *KubernetesNodeController) syncKubernetesNode(key string) (err error)
 
 	kubeNode, err := knc.ds.GetKubernetesNode(name)
 	if err != nil {
-		if datastore.ErrorIsNotFound(err) {
-			logrus.Errorf("Kubernetes node %v has been deleted", key)
-			return nil
+		if !datastore.ErrorIsNotFound(err) {
+			return err
 		}
-		return err
+		logrus.Errorf("Kubernetes node %v has been deleted", key)
+	}
+
+	if kubeNode == nil || kubeNode.DeletionTimestamp != nil {
```

Comment:
Should we count `DeletionTimestamp != nil`? What if the manager is still running on the node?

Also, do we need to do the replica cleanup after the node deletion?

Reply:
OK. Maybe I cannot count `DeletionTimestamp != nil`. And there are some issues/corner cases for the replicas on the node. I will need to re-check those cases.